src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -2733,6 +2733,7 @@ class SetClauseSegment(BaseSegment):
             Ref("BareFunctionSegment"),
             Ref("FunctionSegment"),
             Ref("ColumnReferenceSegment"),
+            Ref("ExpressionSegment"),
             "DEFAULT",
```

Comment:
I'm curious if the addition of this would let us remove some or all of the more specific segment types above.